src/modules/land_detector/LandDetector.h

```diff
@@ -142,38 +140,45 @@ class LandDetector : public ModuleBase<LandDetector>, ModuleParams, px4::Schedul
 	/** Run main land detector loop at this interval. */
 	static constexpr uint32_t LAND_DETECTOR_UPDATE_INTERVAL = 20_ms;
 
-	orb_advert_t _land_detected_pub{nullptr};
-	vehicle_land_detected_s _land_detected{};
-
-	LandDetectionState _state{LandDetectionState::LANDED};
-
 	systemlib::Hysteresis _freefall_hysteresis{false};
 	systemlib::Hysteresis _landed_hysteresis{true};
 	systemlib::Hysteresis _maybe_landed_hysteresis{true};
 	systemlib::Hysteresis _ground_contact_hysteresis{true};
 	systemlib::Hysteresis _ground_effect_hysteresis{false};
 
-	actuator_armed_s _actuator_armed {};
+	actuator_armed_s         _actuator_armed{};
+	vehicle_acceleration_s   _vehicle_acceleration{};
+	vehicle_land_detected_s _land_detected = {
+		.timestamp = 0,
+		.alt_max = -1.0f,
+		.freefall = false,
+		.ground_contact = true,
+		.maybe_landed = true,
+		.landed = true,
+	};
+	vehicle_local_position_s _vehicle_local_position{};
 
-private:
+	orb_advert_t _land_detected_pub{nullptr};
```

Comment:
Can you use the newer API her?

Reply:
Sure, I'll check and add a commit for that.

Reply:
I replaced it and SITL tested again, still fine.